sdks/python/apache_beam/utils/retry.py

```diff
@@ -104,6 +104,15 @@ def retry_on_server_errors_filter(exception):
   return not isinstance(exception, PermanentException)
 
 
+# TODO(BEAM-6202): Dataflow returns 404 for job ids that actuall exist.
+# Retry on those errors.
+def retry_on_server_errors_and_notfound_filter(exception):
```

Comment:
Not for this PR: these retry filters should probably follow a builder pattern.

Reply:
Ack. Thank you for the review.